rheem-platforms/rheem-giraph/src/main/java/org/qcri/rheem/giraph/operators/GiraphPageRankOperator.java

```diff
@@ -0,0 +1,258 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * PageRank {@link Operator} implementation for the {@link GiraphPlatform}.
+ */
+public class GiraphPageRankOperator extends PageRankOperator implements GiraphExecutionOperator {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private String path_out;
+
+    public GiraphPageRankOperator(Integer numIterations) {
+        super(numIterations);
+        setPathOut(null, null);
+    }
+    public GiraphPageRankOperator(PageRankOperator pageRankOperator) {
+        super(pageRankOperator);
+        setPathOut(null, null);
+    }
+
+    @Override
+    public Tuple<Collection<ExecutionLineageNode>, Collection<ChannelInstance>> execute(
+            ChannelInstance[] inputChannelInstances,
+            ChannelInstance[] outputChannelInstances,
+            GiraphExecutor giraphExecutor,
+            OptimizationContext.OperatorContext operatorContext) {
+        assert inputChannelInstances.length == this.getNumInputs();
+        assert outputChannelInstances.length == this.getNumOutputs();
+
+        final FileChannel.Instance inputChannel = (FileChannel.Instance) inputChannelInstances[0];
+        final StreamChannel.Instance outputChanne = (StreamChannel.Instance) outputChannelInstances[0];
+        try {
+            return this.runGiraph(inputChannel, outputChanne, giraphExecutor, operatorContext);
+        } catch (IOException e) {
+            throw new RheemException(String.format("Running %s failed.", this), e);
+        } catch (URISyntaxException e) {
+            throw new RheemException(e);
+        } catch (InterruptedException e) {
+            throw new RheemException(e);
+        } catch (ClassNotFoundException e) {
+            throw new RheemException(e);
+        }
+    }
+
+    private Tuple<Collection<ExecutionLineageNode>, Collection<ChannelInstance>> runGiraph(
+            FileChannel.Instance inputFileChannelInstance,
+            StreamChannel.Instance outputChannelInstance,
+            GiraphExecutor giraphExecutor,
+            OptimizationContext.OperatorContext operatorContext)
+            throws IOException, URISyntaxException, ClassNotFoundException, InterruptedException {
+
+        assert inputFileChannelInstance.wasProduced();
+        Configuration configuration = operatorContext.getOptimizationContext().getConfiguration();
+        String tempDirPath = this.getPathOut(configuration);
+
+        PageRankParameters.setParameter(PageRankParameters.PageRankEnum.ITERATION, this.getNumIterations());
+
+
+        FileSystem fs = FileSystems.getFileSystem(tempDirPath).orElseThrow(
+                () -> new RheemException(String.format("Cannot access file system of %s.", tempDirPath))
+        );
+
+        fs.delete(tempDirPath, true);
```

Comment:
I would say that it is potentially dangerous to delete the output path. What if there is some other Giraph result in there that we still need for some other operator. It would be better to create a new directory in the temp folder for each execution.